src/linear_algebra/basis.lean

```diff
@@ -243,6 +243,35 @@ of_repr (f.symm.trans b.repr)
 
 end map
 
+section map_coeffs
+
+variables {R' : Type*} [semiring R'] [module R' M] (f : R ≃+* R') (h : ∀ c (x : M), f c • x = c • x)
+
+include f h
+
+/-- If `R` and `R'` are isomorphic rings that act identically on a module `M`,
+then a basis for `M` as `R`-module is also a basis for `M` as `R'`-module. -/
+@[simps {simp_rhs := tt}]
+def map_coeffs : basis ι R' M :=
+of_repr { map_smul' := λ c x,
+  begin
+    ext i,
+    rw [← f.apply_symm_apply c, h],
+    simp only [finsupp.map_range.add_equiv_apply, linear_equiv.coe_to_add_equiv, finsupp.coe_smul,
+              add_equiv.coe_trans, function.comp_app, add_equiv.to_fun_eq_coe, smul_eq_mul,
+              finsupp.map_range_apply, pi.smul_apply, ring_equiv.to_add_equiv_eq_coe,
+              ring_equiv.coe_to_add_equiv, linear_equiv.map_smul, ring_equiv.map_mul]
+  end
+  .. b.repr.to_add_equiv.trans (finsupp.map_range.add_equiv f.to_add_equiv) }
```

Comment:
Here's a pretty out there alternative proof:
```
include b
/-- If `R` and `R'` are isomorphic rings that act identically on a module `M`,
then a basis for `M` as `R`-module is also a basis for `M` as `R'`-module. -/
@[simps {simp_rhs := tt}]
def map_coeffs : basis ι R' M :=
let inst : module R' R := module.comp_hom R (↑f.symm : R' →+* R) in by exactI (
have sc : is_scalar_tower R' R R := { smul_assoc := λ x, mul_assoc _ },
have sc : is_scalar_tower R' R M := { smul_assoc := λ x y z, begin
  dsimp [(•)],
  rw [mul_smul, ←h, f.apply_symm_apply],
end },
by exactI (of_repr $ (b.repr.restrict_scalars R').trans $ finsupp.map_range.linear_equiv begin
  refine {map_smul' := λ r' r, _, .. (↑f : _ ≃+ _)},
  apply f.symm.injective,
  simp,
  refl,
end))
```
Perhaps there are components of this that make sense to extract as standalone definitions?

Reply:
Something like that sounds good, although I don't know about `module.comp_hom`, wouldn't that be nicer as `ring_hom.to_algebra f.symm`?

Reply:
`algebra` isn't imported yet as far as I could tell.

Reply:
Added the first `sc` instance in #8707

Reply:
I think I'll keep the hom-based proof over the `scalar_tower`-based proof, because there's a lot less "scary" machinery. But #8707 is a good idea of course.

Reply:
Are you ok to wait for #8707 just to see how much scary machinery ends up left over? Or is this holding up other PRs?

Reply:
ATM only #8701 depends on this PR, and soon I'll be PR'ing some more results based on it. But even if we only get this merged after a month, that'll be fine. Rebasing on top of master is not so bad.